src/motorError.cpp

```diff
@@ -0,0 +1,84 @@
+#include "motorError.h"
+
+Timeout evaluationTimeout;
+Timeout resetTimeout;
+
+InterruptIn motorErrorPin(D0);
+
+#define SEQUENCE_DELAY 1000ms
+#define RESET_DELAY 3000ms
+
+// The error type, determined based on input pulses
+ErrorType errorType = None;
+
+// The pulse count, incremented with every rising edge of the motor error pin
+int pulses = 0;
```

Comment:
nit: make static and volatile since it is a global only being used within this file and multiple interrupts interact with it